### ISSUE TITLE
No Longer a method to extract Activity Log for a resource

### ISSUE THREAD

**Issue by tangentToNothing** [Azure/azure-sdk-for-python#1841]
The `azure-monitor` python package contained the `MonitorClient`, capable of accessing `activity_logs` and filtering by resource_id, resource_group, etc. With `azure-monitor`'s deprecation, surely `azure-mgmt-monitor` would have this type of functionality? Nope. 

Even better, the hosted docs on microsoft's website state that the `MonitorManagementClient` of the `azure-mgmt-monitor` package _should_ be able to fetch `activity_logs`: [link](https://docs.microsoft.com/en-us/python/api/azure.mgmt.monitor.monitormanagementclient?view=azure-python). What gives microsoft? Why support so many language specific SDKs and keep them essentially out of sync with one another _and_ the PS cmdlets?

Just for fun, here is the output of `dir` on a MonitorManagementClient instantiated with the `azure.mgmt` modules on Python 2.7:

`[...<python_specific_outputs>..., 'activity_log_alerts', 'alert_rule_incidents', 'alert_rules', 'autoscale_settings','config','log_profiles','service_diagnostic_settings']`

Honestly, would love a Linus Torvalds-like audit of this repo, this is fourth inconsistency I've run into in a week's time between docs, code samples, the actual code, and the nuances of interfacing with sovereign clouds vs commercial.

**Comment by lmazuel** [comment]
Hi @tangentToNothing 

Let's try to separate all your concerns:

1- Yes, azure-mgmt-monitor has the activity logs. Here's my testing live, no hack:
```shell
(MonitortALTest) D:\VEnvs> pip install azure-mgmt-monitor
Collecting azure-mgmt-monitor
  Using cached azure_mgmt_monitor-0.4.0-py2.py3-none-any.whl
Collecting azure-common~=1.1.6 (from azure-mgmt-monitor)
  Using cached azure_common-1.1.8-py2.py3-none-any.whl
Collecting msrestazure~=0.4.8 (from azure-mgmt-monitor)
  Downloading msrestazure-0.4.21-py2.py3-none-any.whl
Collecting azure-mgmt-nspkg>=2.0.0 (from azure-mgmt-monitor)
  Using cached azure_mgmt_nspkg-2.0.0-py2.py3-none-any.whl
Collecting azure-nspkg>=2.0.0 (from azure-common~=1.1.6->azure-mgmt-monitor)
  Using cached azure_nspkg-2.0.0-py2.py3-none-any.whl
Collecting adal~=0.5.0 (from msrestazure~=0.4.8->azure-mgmt-monitor)
  Downloading adal-0.5.0-py2.py3-none-any.whl (50kB)
    100% |████████████████████████████████| 51kB 1.3MB/s
Collecting msrest<2.0.0,>=0.4.25 (from msrestazure~=0.4.8->azure-mgmt-monitor)
  Downloading msrest-0.4.26-py2.py3-none-any.whl (44kB)
    100% |████████████████████████████████| 51kB 3.4MB/s
Collecting keyring>=5.6 (from msrestazure~=0.4.8->azure-mgmt-monitor)
  Downloading keyring-11.0.0-py2.py3-none-any.whl
Collecting PyJWT>=1.0.0 (from adal~=0.5.0->msrestazure~=0.4.8->azure-mgmt-monitor)
  Using cached PyJWT-1.5.3-py2.py3-none-any.whl
Collecting requests>=2.0.0 (from adal~=0.5.0->msrestazure~=0.4.8->azure-mgmt-monitor)
  Using cached requests-2.18.4-py2.py3-none-any.whl
Collecting python-dateutil>=2.1.0 (from adal~=0.5.0->msrestazure~=0.4.8->azure-mgmt-monitor)
  Using cached python_dateutil-2.6.1-py2.py3-none-any.whl
Collecting cryptography>=1.1.0 (from adal~=0.5.0->msrestazure~=0.4.8->azure-mgmt-monitor)
  Using cached cryptography-2.1.4-cp36-cp36m-win_amd64.whl
Collecting isodate>=0.6.0 (from msrest<2.0.0,>=0.4.25->msrestazure~=0.4.8->azure-mgmt-monitor)
  Using cached isodate-0.6.0-py2.py3-none-any.whl
Collecting certifi>=2017.4.17 (from msrest<2.0.0,>=0.4.25->msrestazure~=0.4.8->azure-mgmt-monitor)
  Using cached certifi-2018.1.18-py2.py3-none-any.whl
Collecting requests-oauthlib>=0.5.0 (from msrest<2.0.0,>=0.4.25->msrestazure~=0.4.8->azure-mgmt-monitor)
  Using cached requests_oauthlib-0.8.0-py2.py3-none-any.whl
Collecting pywin32-ctypes!=0.1.0,!=0.1.1; sys_platform == "win32" (from keyring>=5.6->msrestazure~=0.4.8->azure-mgmt-monitor)
Collecting urllib3<1.23,>=1.21.1 (from requests>=2.0.0->adal~=0.5.0->msrestazure~=0.4.8->azure-mgmt-monitor)
  Using cached urllib3-1.22-py2.py3-none-any.whl
Collecting idna<2.7,>=2.5 (from requests>=2.0.0->adal~=0.5.0->msrestazure~=0.4.8->azure-mgmt-monitor)
  Using cached idna-2.6-py2.py3-none-any.whl
Collecting chardet<3.1.0,>=3.0.2 (from requests>=2.0.0->adal~=0.5.0->msrestazure~=0.4.8->azure-mgmt-monitor)
  Using cached chardet-3.0.4-py2.py3-none-any.whl
Collecting six>=1.5 (from python-dateutil>=2.1.0->adal~=0.5.0->msrestazure~=0.4.8->azure-mgmt-monitor)
  Using cached six-1.11.0-py2.py3-none-any.whl
Collecting asn1crypto>=0.21.0 (from cryptography>=1.1.0->adal~=0.5.0->msrestazure~=0.4.8->azure-mgmt-monitor)
  Using cached asn1crypto-0.24.0-py2.py3-none-any.whl
Collecting cffi>=1.7; platform_python_implementation != "PyPy" (from cryptography>=1.1.0->adal~=0.5.0->msrestazure~=0.4.8->azure-mgmt-monitor)
  Using cached cffi-1.11.4-cp36-cp36m-win_amd64.whl
Collecting oauthlib>=0.6.2 (from requests-oauthlib>=0.5.0->msrest<2.0.0,>=0.4.25->msrestazure~=0.4.8->azure-mgmt-monitor)
Collecting pycparser (from cffi>=1.7; platform_python_implementation != "PyPy"->cryptography>=1.1.0->adal~=0.5.0->msrestazure~=0.4.8->azure-mgmt-monitor)
Installing collected packages: azure-nspkg, azure-common, PyJWT, certifi, urllib3, idna, chardet, requests, six, python-dateutil, asn1crypto, pycparser, cffi, cryptography, adal, isodate, oauthlib, requests-oauthlib, msrest, pywin32-ctypes, keyring, msrestazure, azure-mgmt-nspkg, azure-mgmt-monitor
Successfully installed PyJWT-1.5.3 adal-0.5.0 asn1crypto-0.24.0 azure-common-1.1.8 azure-mgmt-monitor-0.4.0 azure-mgmt-nspkg-2.0.0 azure-nspkg-2.0.0 certifi-2018.1.18 cffi-1.11.4 chardet-3.0.4 cryptography-2.1.4 idna-2.6 isodate-0.6.0 keyring-11.0.0 msrest-0.4.26 msrestazure-0.4.21 oauthlib-2.0.6 pycparser-2.18 python-dateutil-2.6.1 pywin32-ctypes-0.1.2 requests-2.18.4 requests-oauthlib-0.8.0 six-1.11.0 urllib3-1.22
(MonitortALTest) D:\VEnvs> ipython
WARNING: Attempting to work in a virtualenv. If you encounter problems, please install IPython inside the virtualenv.
Python 3.5.4 (v3.5.4:3f56838, Aug  8 2017, 02:07:06) [MSC v.1900 32 bit (Intel)]
Type "copyright", "credits" or "license" for more information.

IPython 4.0.1 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.

In [1]: from azure.mgmt.monitor import MonitorManagementClient

In [2]: client = MonitorManagementClient("", "")

In [4]: client.activity_logs
Out[4]: <azure.mgmt.monitor.operations.activity_logs_operations.ActivityLogsOperations at 0x6465c10>
```

Please double-check your installation that you have at least `azure-mgmt-monitor` 0.4.0

2- "Why support so many language specific SDKs and keep them essentially out of sync with one another and the PS cmdlets?"

That's essentially not true for Python, since CLI is in Python and CLI and Powershell are in sync. If unfortunately we were not able to keep the pace in a feature you need, just open an issue to describe the feature you see in Powershell that is not yet in Python and we will prioritize it. 
For other language, same applies, just open an issue if the SDK team of this language is not up-to-date with something you need.
PS is built top of a meta-description of the RestAPI, meaning that if PS has it, all other language can have it as well. It's just a matter of prioritizing it.

3- "fourth inconsistency", please open issue for any inconsistency so we can fix them. We're definitely not perfect, please point us to the actual issues you found so we can fix them and provide better experience,

Thank you,

**Comment by tangentToNothing** [comment]
Closing this, I addressed a lot of these through various other channels, a lot of it through Azure support and some customer reps.

My most sincere apologies for being quite rude in my issue posting. I'm sincerely impressed by your calm and thorough response to me!

One of the key things that helped me out: my pointer package `azure` was too old, and the organization I am in had an in-house package index that failed to update against PyPi.

Here's to a better, nicer internet I guess.
